### ISSUE TITLE
🐞[BUG]: Type error with custom router states

### ISSUE THREAD

**Issue by hpop** [ngxs/store#2264]
### Affected Package

The issue is caused by package @ngxs/router-plugin

### Is this a regression?

Yes, the previous version in which this bug was not present was: 18.1.4

### Description
In 18.1.4 it was possible to type the `RouterState.state` function via generics. 
In 18.1.5 This is not possible anymore.

## 🔬 Minimal Reproduction
```
this.store.selectSnapshot(RouterState.state<CustomRouterStateParams>);
```

## 🔥 Exception or Error
<pre><code>

✘ [ERROR] TS2345: Argument of type '{}' is not assignable to parameter of type 'TypedSelector<RouterStateParams>'. [plugin angular-compiler]

***.ts:72:40:
      72 │ ...this.store.selectSnapshot(RouterState.state&lt;RouterStateParams&gt;)!);
         ╵                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

</code></pre>

## Environment

<pre><code>
Libs:
- @angular/core version: 19.0.0
- @ngxs/store version: 18.1.5

For Tooling issues:
- Node version: 23
- Platform:  Mac

Others:

</code></pre>

**Comment by arturovt** [comment]
I have released v18.1.6 and reverted the commit that caused that issue.

**Comment by hpop** [comment]
Thanks a lot. It's fixed for me!
